content/en/docs/components/pipelines/sdk/component-development.md

```diff
@@ -1,85 +1,93 @@
 
-When creating a component you need to think about how the component
-will communicate with upstream and downstream components. That is, how it will consume
-the input data and produce the output data.
+A pipeline component is a self-contained set of code that performs one step in
+your ML workflow. This document describes the concepts required to build
+components, and demonstrates how to get started building components.
 
-### Producing data
+## Before you begin
 
-To output any piece of data, the component program must write the output data to some location and inform the system
-about that location so that the system can pass the data between steps.
-The program should accept the paths for the output data as command-line arguments. That is, you should not hardcode the paths.
+Run the following command to install the Kubeflow Pipelines SDK.
 
-In practical terms this means that the program should receive local paths for every output the component produces and write data to those paths.
-Usually an output is a single piece of data. In this case the program should treat the given output path as a file path, open it for writing and write the data there.
-In some cases there is a need to output a directory of files. In this case the program should treat the given output path as a directory path, create a directory at that location and then add files to that directory. In both cases it might be necessary to create parent directories if they do not exist.
-
-#### Advanced: Producing data in an external system
-
-In some scenarios, the goal of the component is to create some data object in an external service (for example a BigQuery table).
-In this case the component program should do that and then output some identifier of the produced data location (for example BigQuery table name) instead of the data itself.
-This scenario should be limited to cases where the data must be put into external system instead of keeping it inside the Kubeflow Pipelines system.
-The class of components where this behavior is common is exporters (for example "upload data to GCS").
-Note that the Pipelines system cannot provide consistency and reproducibility guarantees for the data outside of its control.
-
-### Consuming data
-
-There are two main ways a command-line program usually consumes data:
-
-* Small pieces of data are usually passed as command-line arguments: `program.py --param1 100`.
-* Bigger data or binary data is usually stored in files and then the file paths are passed to the program: `program.py --data1 /inputs/data1.txt`. The system needs to be aware about the need to put some data into files and pass their paths to the program.
-
-## Writing the program code
-
-This section describes an example program that has two inputs (for small and
-large pieces of data) and one output. Although the programming language in this example
-is Python 3, you can build your component in any language.
+```python
+$ pip3 install kfp --upgrade
```

Comment:
The component building part of the guide does not need `kfp`. KFP only used near the end in a dummy example pipeline.

Reply:
Since users will need to install KFP SDK to complete the guide, I would prefer to keep this section.